vb-fe/src/util/Http.ts

```diff
@@ -11,8 +14,80 @@ export enum HttpStatus {
   InternalError = 500,
 }
 
-// TODO add access token and refresh token to headers
+export enum TokenStorageKeys {
+  AT = "v-at",
+  RT = "v-rt",
+  SpotifyAT = "v-s-at",
+  SpotifyRT = "v-s-rt",
+}
+
+interface RTResponse {
+  accessToken: string;
+  refreshToken: string;
+  spotifyAccessToken: string;
+}
+
+export const buildHeaders = () => ({
+  [TokenStorageKeys.AT]: localStorage.getItem(TokenStorageKeys.AT) || "",
+  [TokenStorageKeys.RT]: localStorage.getItem(TokenStorageKeys.RT) || "",
+  [TokenStorageKeys.SpotifyAT]: localStorage.getItem(TokenStorageKeys.SpotifyAT) || "",
+  [TokenStorageKeys.SpotifyRT]: localStorage.getItem(TokenStorageKeys.SpotifyRT) || "",
+});
+
+const setHeadersToLocalStorage = (
+  accessToken: string,
+  refreshToken: string,
+  spotifyAccessToken: string,
+  spotifyRefreshToken: string,
+) => {
+  localStorage.setItem(TokenStorageKeys.AT, accessToken);
+  localStorage.setItem(TokenStorageKeys.RT, refreshToken);
+  localStorage.setItem(TokenStorageKeys.SpotifyAT, spotifyAccessToken);
+  localStorage.setItem(TokenStorageKeys.SpotifyRT, spotifyRefreshToken);
+};
+
 export const Http = axios.create({
   baseURL: BASE_URL,
   withCredentials: true,
 });
+
+export const setHeaders = () => {
+  Http.defaults.headers.common = buildHeaders();
+};
+
+export const initHttp = () => {
+  setHeaders();
+
+  Http.interceptors.response.use(
+    (value: AxiosResponse): AxiosResponse<any> | Promise<AxiosResponse<any>> => {
+      return value;
+    },
+    (err: AxiosError): Promise<any> => {
+      return new Promise((resolve, reject) => {
+        const originalRequest = err.config;
+        const { retry } = store.getState().system;
+
+        if (err.response?.status === HttpStatus.Unauthorized && !retry) {
```

Comment:
We should add in functionality so that if this retry fails, we clear localStorage and redirect to the login page. Thoughts?

Reply:
yep you're right. let me do that